packages/xod-client-electron/src/view/containers/App.jsx

```diff
@@ -147,6 +148,23 @@ class App extends client.App {
       }
     );
 
+    this.urlActions = {
+      // actionPathName: params => this.props.actions.someAction(params.foo, params.bar),
+      [client.URL_ACTION_TYPES.OPEN_TUTORIAL]: this.onOpenTutorialProject,
+    };
+    ipcRenderer.on(
+      EVENTS.XOD_URL_CLICKED,
+      (event, { actionName, params }) => {
+        const action = this.urlActions[actionName];
+
+        if (action) {
+          action(params);
+        } else {
+          this.props.actions.addError({ title: 'Invalid action name' });
```

Comment:
Such phrases should reside in `messages.js` (`packages/xod-client-electron/src/shared/messages.js` in this case). Suggest:

```js
// messages.js
const invalidActionName = action => composeMessage(
  'Bad link',
  `XOD link points to an unknown action '${action}'`
);

// App.jsx
this.props.actions.addError(Messages.invalidActionName(action));
```